library/src/updater.rs

```diff
@@ -268,6 +268,10 @@ pub fn check_for_downloadable_update(channel: Option<&str>) -> anyhow::Result<bo
     let response = request_fn(&url, request)?;
     shorebird_debug!("Patch check response: {:?}", response);
 
+    if let Some(rolled_back_patches) = response.rolled_back_patch_numbers {
+        roll_back_patches_if_needed(rolled_back_patches)?;
```

Comment:
Do we have any deadlock risk from this?  This is grabbing a lock now in a pathway it wasn't before?

Reply:
Discussed in person, but I don't think so. This should be a no-op for the standard update pathway, and check_for_downloadable_update doesn't hold on to a config lock.